### DISCUSSION TITLE
Maintaining Stable Use Cases on Swan Chain

### DISCUSSION THREAD

**Discussion by Normalnoise** [swanchain/governance#9] [Proposals]
## Objective  
To showcase the availability and reliability of Swan Chain's computing resources (Computing Providers), this project aims to:  
1. Demonstrate the diverse computational tasks that Swan Chain can execute.  
2. Highlight its capability to support various types of computational tasks and application services.  

---

## Rationale  
- **Testing Utility:** Validate Swan Chain's performance and reliability in handling computational tasks.  
- **Market Impact:** Strengthen Swan Chain’s reputation in the market by showcasing real-world use cases.  
- **Innovation:** Encourage users to submit deployment examples for computational tasks they wish to see in action.  

---

## Implementation Plan  

1. **Selection of Initial Use Cases:**  
   - The first batch of use cases will be chosen from [Lagrange‘s Use Case List](https://docs.lagrangedao.org/spaces/use-case).  
   - Use cases will be run and demonstrated via the Lagrange platform.  

2. **Stable Computing Provider Selection:**  
   - Identify computing providers capable of offering stable services.  
   - Open applications to all providers willing to undertake computational tasks.  

3. **Budget:**  
   - **Total Budget:** 253,000 SWAN tokens.  
   - **Duration:** 6 months.  
   - **Wallet Address:** `0x6D68A3d5B4bcB86215179e2604846Db89cE67681`
   
### Budget Plan:
 As the Usecase Lists show, at least 14 applications will be deployed on Swan Chain's CPs. For different application needs different instances, we plan to deploy them to these instances:
 - G1ae.small(Nvidia 3080 · 4 vCPU · 8 GiB)   1.84 SWAN per hour
 - G1ae.medium(Nvidia 3080 · 8 vCPU · 16 GiB) 2.08 SWAN per hour
 - R1ae.medium(Nvidia 3080 · 8 vCPU · 32 GiB) 2.4 SWAN per hour
 - 2*T1ae.3xlarge(2 Nvidia 3080 · 12 vCPU · 64 GiB)  6.24 SWAN per hour
 - 2*Hpc1ae.xlarge(2 Nvidia 3090 Ti · 8 vCPU · 16 GiB)  6.02 SWAN per hour
 - 2*T1az.2xlarge(2 Nvidia 4090 · 8 vCPU · 64 GiB).     9.88 SWAN per hour
 - 2*T1az.3xlarge(2 Nvidia 4090 Ti · 8 vCPU · 64 GiB)  9.48 SWAN per hour
Maybe some applications need higher GPU instances, we still will deploy them, so the budget is only a rough estimation.

### Cost Calculation

Consisering the SWAN token price and the GPU's renting price, we calculate the total costs using an average price of **4.17 SWAN per hour**. The example details are here:

Taking the market rental price of 3080 as reference, 
 - now, renting price is $0.25 /hour, 
 - SWAN token price is $0.06, 
 - roughly converted, we get `0.25/0.06 = 4.17 SWAN/hour`

##### Parameters:
- **Number of applications**: 14
- **Duration**: 6 months  
`6 months * 30 days/month * 24 hours/day = 4,320 hours`
- **Average price per hour**: 4.17 SWAN
- In consideration of possible losses due to redeployment, this is roughly stated as `800 SWAN`

#### Total Cost:
The formula is:  
`Total cost = Number of applications * Duration (hours) * Price per hour + possible losses`

Substitute the values:  
`Total cost = 14 * 4320 * 4.17 + 800= 253,000 SWAN`

**Total Cost**: **253,000 SWAN**

4. **Risk Assessment:**  
   - **Market Volatility:** Token price fluctuations may impact the project's funding adequacy.  
   - **Provider Stability:** Instability in a few computing providers may result in certain use cases becoming temporarily unavailable.  
   - **Other Risks:** Unforeseen technical or operational challenges.  

5. **Next Steps:**  
   - **User Feedback Engagement:**  
     - Regularly gather insights from the community through interactive discussions or surveys.  
     - Focus on identifying user needs and ideas for future improvements to use cases.  
   - **Iterative Updates:**  
     - Periodically review user and market feedback to prioritize actionable changes to use cases.  
     - Implement and test improvements or adjustments to enhance reliability and relevance.  
   - **Provider Performance Evaluation:**  
     - Continuously monitor computing provider performance, focusing on stability and reliability.  
     - Share periodic performance highlights or issues with the community to maintain transparency.  

---

This initiative will reinforce Swan Chain’s potential as a reliable and innovative computing resource provider, fostering broader adoption and user engagement.

**Comment by CryptoSenorita** [comment]
Hi @Normalnoise , 

We are evaluating this proposal, thank you for your patience. Regarding the payment cycle, can we pay once every 2 months?

The Swan Chain Team

**Reply by Normalnoise** [reply]
@CryptoSenorita great, I think it is ok; Let's keep pushing forward

**Reply by CryptoSenorita** [reply]
Ok. The token transfer schedule will be:
2025.1.10: 84,333.33 Swan Tokens (has been transferred)
2025.3.10: 84,333.33 Swan Tokens
2025.5.10: 84,333.34 Swan Tokens
253,000 Swan Tokens in total.

**Comment by CryptoSenorita** [comment]
Hi @Normalnoise,

Pls check that the first 2 payments have already been paid and the last one will be scheduled on May 10th. Thank you.

The Swan Chain Team

**Reply by Normalnoise** [reply]
Yes, received

**Comment by Normalnoise** [comment]
@CryptoSenorita 
The use case has demonstrated strong operational performance during its initial phase, validating both the deployment framework and its practical value. However, due to a significant drop in token price, the original budget is no longer sufficient to cover the ongoing costs of computing providers (CPs), which has put continuous deployment and service operations at risk.

We are applying for additional funding to maintain service stability and ensure completion of the planned phase. Based on current estimates, the daily operating cost is approximately **38,650 SWAN**. With around **60 days** remaining until the scheduled end of the use case, the **total additional funding required is approximately 2.319 million SWAN**.

We appreciate your understanding and support as we secure the resources necessary to sustain the project’s momentum and outcomes.
